public/js/replay.js

```diff
@@ -70,23 +94,169 @@ function Replay (params) {
+    instance.current_v = version;
+    instance.time_slider.slider ('setValue', instance.current_v);
+    instance.setCurrentFlag();
+    return 'Version: ' + version;
+  };
+
+
+  /*
+  * Moves to next flag
+  */
+  this.nextFlag = function(){
+    if (!this.stopped){
+      $('#start-replay-button').toggleClass('active');
+      $('#start-replay-button').find('span.glyphicon').toggleClass('glyphicon-pause').toggleClass('glyphicon-play');
+      this.stop();
+    }
+    this.at_flag = true;
+    this.setVersion(this.flagged[this.next_flag]);
+
+  };
+
+  /*
+  * Moves to previous flag
+  */
+  this.prevFlag = function(){
+    if(!this.stopped){
+      $('#start-replay-button').toggleClass('active');
+      $('#start-replay-button').find('span.glyphicon').toggleClass('glyphicon-pause').toggleClass('glyphicon-play');
+      this.stop();
+    }
+    this.at_flag = true;
+    this.setVersion(this.flagged[this.prev_flag]);
+
+
+
+
+  };
+
+  /*
+  * Gets the current/prev/next flag based on the position of the slider
+  * Iterates through all flags, each time the slider moves, so for a large
+  * number of flagged versions, it may have performance issues.
+  */
+  this.setCurrentFlag = function(){
+    var index = 0;
+    if (this.current_v < this.flagged[0]){
+      this.current_flag = null;
+      this.prev_flag = null;
+      this.next_flag = 0;
+    }
+    else if(this.current_v === this.flagged[0]){
+      this.current_flag = 0;
+      this.prev_flag = null;
+      this.next_flag = 1;
+    }
+    else if(this.current_v > this.flagged[this.flagged.length-1]){
+      this.current_flag = this.flagged.length-1;
+      this.prev_flag = this.flagged.length-1;
+      this.next_flag = null;
+    }
+    else if(this.current_v === this.flagged[this.flagged.length-1]){
+      this.current_flag = this.flagged.length-1;
+      this.prev_flag = this.flagged.length-2;
+      this.next_flag = null;
+    }
+    else{
+      while(index < this.flagged.length-1){
+        if (this.current_v > this.flagged[index] && this.current_v < this.flagged[index+1]){
+          this.current_flag = index;
+          this.prev_flag = index;
+          this.next_flag = index+1;
+          break;
+        }
+        else if(this.current_v === this.flagged[index]){
+          this.current_flag = index;
+          this.prev_flag = index - 1;
+          this.next_flag = index + 1;
+          break;
+        }
+        index++;
+      }
+
+    }
+    this.checkFlagButtons();
   };
 
+  /*
+  * Checks to see if the left/right flag buttons should be disabled. If
+  * there are no more flags on either side, disable it.
+  */
+  this.checkFlagButtons = function(){
+    if(this.prev_flag !== null){
+      $('#flag-left').prop("disabled",false);
+    }else{
+      $('#flag-left').prop("disabled",true);
+    }
+    if(this.next_flag !== null){
+      $('#flag-right').prop("disabled",false);
+    }
+    else{
+      $('#flag-right').prop("disabled",true);
+    }
+
```

Comment:
I like Mark's suggestion with a tweak - I think it's more readable if you invert the logic. `$('flag-right').prop("disabled", this.next_flag === null)` means "set disabled (to true) if next_flag is null"

And do the same thing for prev_flag